pkg/component/gardenercontrollermanager/configmaps.go

```diff
@@ -0,0 +1,168 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gardenercontrollermanager
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	componentbaseconfigv1alpha1 "k8s.io/component-base/config/v1alpha1"
+	"k8s.io/utils/pointer"
+
+	controllermanagerv1alpha1 "github.com/gardener/gardener/pkg/controllermanager/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/logger"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+const (
+	configMapControllerManagerPrefix = "gardener-controller-manager-config"
+	dataConfigKey                    = "config.yaml"
+)
+
+var controllerManagerCodec runtime.Codec
+
+func init() {
+	controllerManagerScheme := runtime.NewScheme()
+	utilruntime.Must(controllermanagerv1alpha1.AddToScheme(controllerManagerScheme))
+
+	var (
+		ser = json.NewSerializerWithOptions(json.DefaultMetaFactory, controllerManagerScheme, controllerManagerScheme, json.SerializerOptions{
+			Yaml:   true,
+			Pretty: false,
+			Strict: false,
+		})
+		versions = schema.GroupVersions([]schema.GroupVersion{
+			controllermanagerv1alpha1.SchemeGroupVersion,
+		})
+	)
+
+	controllerManagerCodec = serializer.NewCodecFactory(controllerManagerScheme).CodecForVersions(ser, ser, versions, versions)
+}
+
+func (g *gardenerControllerManager) configMapControllerManagerConfig() (*corev1.ConfigMap, error) {
+	controllerManagerConfig := &controllermanagerv1alpha1.ControllerManagerConfiguration{
+		GardenClientConnection: componentbaseconfigv1alpha1.ClientConnectionConfiguration{
+			QPS:        100,
+			Burst:      130,
+			Kubeconfig: gardenerutils.PathGenericKubeconfig,
+		},
+		Controllers: controllermanagerv1alpha1.ControllerManagerControllerConfiguration{
+			CertificateSigningRequest: &controllermanagerv1alpha1.CertificateSigningRequestControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(5),
+			},
+			ControllerRegistration: &controllermanagerv1alpha1.ControllerRegistrationControllerConfiguration{
```

Comment:
Not related to this change, but `.Values.global.controller.config.controllers.controllerRegistration.concurrentSyncs` is a required field in the chart, but is not present in `charts/gardener/controlplane/charts/runtime/values.yaml`, Can you add that?

Reply:
Same for secretbinding config.

Reply:
I am not sure I get your point as those attributes are actually not required:

```
      {{- if .Values.global.controller.config.controllers.controllerRegistration }}
      controllerRegistration:
        concurrentSyncs: {{ required ".Values.global.controller.config.controllers.controllerRegistration.concurrentSyncs is required" .Values.global.controller.config.controllers.controllerRegistration.concurrentSyncs }}
      {{- end }}
```

`concurrentSyncs` is only required if `controllerRegistration` is defined.

I added both `controllerRegistration` and `secretBinding` including `concurrentSyncs` to `values.yaml` anyway as it does not cause any harm to have them.

Change will be included in the next commit.